crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1963,6 +1969,29 @@ fn prim_ty_to_bty(prim_ty: rustc_hir::PrimTy) -> rty::BaseTy {
 
 /// Conversion of expressions
 impl<'genv, 'tcx: 'genv, P: ConvPhase<'genv, 'tcx>> ConvCtxt<P> {
+    fn conv_lit(&self, lit: fhir::Lit, fhir_id: FhirId, span: Span) -> QueryResult<rty::Constant> {
+        match lit {
+            fhir::Lit::Int(n) => {
+                if let Some(rty::Sort::BitVec(rty::BvSize::Fixed(size))) =
+                    self.results().literal_sort(fhir_id)
+                {
+                    debug_assert!(size == 32 || size == 64);
+                    if 0 <= n && n < (1 << size) {
```

Comment:
This also could overflow. I think the following is equivalent, and it doesn't overflow. 

```suggestion
                    if 0 <= n && (n == 0 || n.ilog2() < size) {
```

note to myself: I don't know how we end up with a `i128` in a literal; there's no way to write negative literals.